pkg/hiveregistrator/registrator.go

```diff
@@ -0,0 +1,164 @@
+package hiveregistrator
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	v1 "github.com/openshift/hive/apis/hive/v1"
+	hiveclient "github.com/openshift/hive/pkg/client/clientset/versioned"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+type Registrator interface {
+	InitializeFromFile(ctx context.Context, hiveKubeConfigPath string) error
+	Register(ctx context.Context) error
+	IsRegistered(ctx context.Context) (bool, error)
+	IsConnected(ctx context.Context) (bool, error)
+}
+
+// Represents all data in hive pertaining to a single ARO cluster
+type WorkloadCluster struct {
+	Subscription                string `json:"subscription,omitempty"`
+	Name                        string `json:"name,omitempty"`
+	Location                    string `json:"location,omitempty"`
+	BaseDomain                  string `json:"basedomain,omitempty"`
+	BaseDomainResourceGroupName string `json:"basedomainResourceGroupName,omitempty"`
+	InfraID                     string `json:"infraId,omitempty"`
+	ClusterID                   string `json:"clusterID,omitempty"`
+	KubeConfig                  string `json:"kubeconfig,omitempty"`
+	PullSecret                  string `json:"pullsecret,omitempty"`
+	ServicePrincipal            string `json:"serviceprincipal,omitempty"`
+}
+
+type hiveRegistrator struct {
+	Hiveclientset   *hiveclient.Clientset
+	Kubernetescli   *kubernetes.Clientset
+	WorkloadCluster *WorkloadCluster
+}
+
+func NewRegistrator(workloadcluster *WorkloadCluster) Registrator {
```

Comment:
Nit: maybe just `New`? So from outside it is going to be `hiveregistrator.New(...)`